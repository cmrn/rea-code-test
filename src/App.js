import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        app go here
      </div>
    );
  }
}

export default App;
