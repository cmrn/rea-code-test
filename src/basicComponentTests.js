import React from 'react';
import { shallow } from 'enzyme';

export default function basicComponentTests(component) {
  it('allows a className to be set', () => {
    const wrapper = shallow(React.cloneElement(component, { className: "classy" }));
    expect(wrapper.props().className).toEqual(expect.stringContaining('classy'));
  });

  it('passes through other props', () => {
    const wrapper = shallow(React.cloneElement(component, { foo: "bar" }));
    expect(wrapper.props().foo).toEqual('bar');
  });
}
