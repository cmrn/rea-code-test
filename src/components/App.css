.App {
  max-width: 800px;
  min-width: 680px;
  margin: 0 auto;
  display: flex;
  justify-content: space-between;
}