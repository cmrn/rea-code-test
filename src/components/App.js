import React from 'react';
import Results from '../containers/Results';
import './App.css';

const App = (props) => (
  <div {...props}>
    <Results />
  </div>
);

export default App;
