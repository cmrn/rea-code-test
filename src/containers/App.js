import React from 'react';
import App from '../components/App';
import { view } from 'react-easy-state';
import propertyStore from '../propertyStore';

class AppContainer extends React.Component {
  componentDidMount() {
    propertyStore.load();
  }

  render() {
    return <App {...this.props} />;
  }
}

export default view(AppContainer);
