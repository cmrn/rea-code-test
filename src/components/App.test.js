import React from 'react';
import basicComponentTests from '../basicComponentTests';
import { shallow } from 'enzyme';
import App from './App';

basicComponentTests(<App />);

it('renders the app with Results and Saved components', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
