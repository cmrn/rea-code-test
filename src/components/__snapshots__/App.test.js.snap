// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app with Results and Saved components 1`] = `
<div
  className="App"
>
  <ResultsContainer />
  <SavedContainer />
</div>
`;
